doc/architecture/architecture.rst

```diff
@@ -1,16 +1,190 @@
+on the `Yocto Project_` and is provided
+as pre-compiled and configured distribution for IoT. There is no
+end-user compilation needed when starting from scratch with Ostro OS
+since the build system is configured to use the shared. In addition,
+pre-compiled reference images are available for download.
+
+3. Ostro OS is secured and built with security in mind: Security is
+critical for IoT and it is also a focus in Ostro OS. The security
+design is to provide scalable options how security can be configured
+in Ostro based products. The security architecture is explained in
+more details in :ref: `system-and-security-architecture`.
+
+4. Ostro OS is easy to customize: While Ostro OS comes pre-compiled
+(binaries and images), it is important that the OS can be customized
+to meet the diverse product needs in IoT space. With the help of
+Yocto Project build system used by Ostro OS, it is easy to customize
+the build and images to suit the needs. :ref:`quick_start/about` and
+the links provided there give a good start to Ostro OS customization.
+
+Ostro |trade| OS Architecture Stack
+===================================
+
+One way to descibe Ostro OS architecture is to build a stack from
+hardware platform up to an IoT Application layer. This stack is
+illustrated in Fig 1.
+
+.. image::images/ostro_os_architecture.png
+
+In the following, each layer is explained in more detail.
+
+IoT Applications
+----------------
+The IoT applications layer consist of all applications that use the
+underlying platform. Ostro OS itself currently does not
+implement any concrete IoT use-cases via applications but many sample
+applications are provided to demonstrate the OS services.
+
+Programming Interfaces
+-----------------------
+Ostro OS provides various application runtimes to allow the end-user
+more choices to use in their IoT applications.
+
+* `Soletta Project`_ is an open source
+framework for making IoT applications. The project provides libraries
+to make it quick and easy to write software for IoT devices. The
+applications can we written using a flow-based programing (FBP) or
+more traditionally as C applications linking to Soletta's C based
+platform APIs.
+
+* `Node.JS`_ is a popular JavaScript runtime to
+run (IoT) web applications. Ostro OS provides the Node.JS runtime
+and selected JavaScript APIs (as Node.JS modules) to build IoT
+applications. The set of APIs include, e.g., a Javascript API for
+the OIC specification. The Javascript API set is kept aligned with
+the API set provided by Soletta.
+
+* In addition to Soletta and Node.JS, common programming languages/
+runtimes in IoT space are supported. Ostro OS provides Python runtime
+and Java Platform (OpenJDK8) that are popular in the IoT space.
+
+Frameworks
+----------
+
+Frameworks provide an abstraction to platform services. For making
+application deployment easy, the Ostro Application
+Framework :ref:`application_framework` provides tooling for
```

Comment:
On Mon, Apr 25, 2016 at 6:59 PM, David Kinder <notifications@github.com>
wrote:

> In doc/architecture/architecture.rst
> <https://github.com/ostroproject/meta-ostro/pull/134#discussion_r60938910>
> :
>
> > +run (IoT) web applications. Ostro OS provides the Node.JS runtime
> > +and selected JavaScript APIs (as Node.JS modules) to build IoT
> > +applications. The set of APIs include, e.g., a Javascript API for
> > +the OIC specification. The Javascript API set is kept aligned with
> > +the API set provided by Soletta.
> > +
> > +* In addition to Soletta and Node.JS, common programming languages/
> > +runtimes in IoT space are supported. Ostro OS provides Python runtime
> > +and Java Platform (OpenJDK8) that are popular in the IoT space.
> > +
> > +Frameworks
> > +----------
> > +
> > +Frameworks provide an abstraction to platform services. For making
> > +application deployment easy, the Ostro Application
> > +Framework :ref:`application_framework` provides tooling for
>
> I went through the document fixing broken :ref: name syntax and getting
> the linking and formatting right and did a PR directly to Mikko’s repo
> rather than inundating this PR with lots of little fixes.
>
>
@dbkinder your changes are updated in this PR too. Thanks! I'll do the same
with Geoffroy's improvements shortly.